JStanley_Oloughlin_Transmissivity_Calculation

```diff
@@ -0,0 +1,448 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Feb  1 09:42:07 2023
+
+@author: josep
+"""
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import math
+
+def baseflow_filter(y, a, BFImax):
+    """
+    Implementation of the Eckhardt 2005 (HP) recursive digital filter for baseflow separation. Courtesy of Dr. Ciaran Harman. 
+    """
+    N = len(y) # length of the timeseries
+    C = (1 - a) * BFImax / (1 - BFImax)
+    A = a / (1 + C)
+    B = C / (1 + C)
+    b = np.zeros(N)
+    b[0] = BFImax * np.mean(y)
+    for k in range(1,N):
+        if np.isnan(y[k]):
+            b[k] = A * b[k-1]
+        else:
+            b[k] = A * b[k-1] + B * y[k]
+        if b[k]>y[k]:
+            b[k] = y[k]
+    return b
+
+def mouse_event(event):
+    """
+    Parameters
+    ----------
+    event : mouse click
+       This is used so that index points can be found by clicking at any point on the graph.
+
+    Returns
+    -------
+    None.
+
+    """
+    print('x: {} and y: {}'.format(event.xdata, event.ydata))
+        
+def baseflow_calc(Q, P):
```

Comment:
Why does this function take P? When it takes Q, it should just take the single column with discharge, since it doesn't use any of the other columns.

Reply:
more generally I don't think I see why this function is needed - it just sets particular values for `baseflow_filter`?